app/controllers/admin/export/document_controller.rb

```diff
@@ -55,4 +59,34 @@ def page_number
   def items_per_page
     params.fetch(:page_count, 100)
   end
+
+  def linked_from_editions(edition)
+    EditionDependency.where(dependable_id: edition.id, dependable_type: "Edition").map(&:edition)
+  end
+
+  def whitehall_admin_links(body)
+    govspeak = build_govspeak_document(body)
+    Govspeak::LinkExtractor.new(govspeak).call
+  end
+
+  def whitehall_admin_link_edition(link)
+    Whitehall::AdminLinkLookup.find_edition(link)
+  end
+
+  def edition_public_url(edition)
+    Whitehall.url_maker.public_document_url(edition)
+  end
+
+  def update_linking_documents(edition)
+    linked_from_editions(edition).each do |linked_from_edition|
+      linked_from_edition.translations.each do |translation|
+        whitehall_admin_links(translation.body).each do |whitehall_admin_link|
+          if whitehall_admin_link_edition(whitehall_admin_link).document_id == edition.document_id
+            translation.body.sub!(whitehall_admin_link, edition_public_url(edition))
+          end
+        end
+        translation.save!
+      end
+    end
```

Comment:
We may want to think about refactoring this as 3 nestings of loop is pretty tough for comprehension.